opsdroid/connector/matrix/connector.py

```diff
@@ -111,100 +109,115 @@ def filter_json(self):
             "room": {"account_data": {"types": []}, "ephemeral": {"types": []}},
         }
 
-    async def make_filter(self, api):
+    async def make_filter(self, api, fjson):
         """Make a filter on the server for future syncs."""
-        fjson = self.filter_json
-        resp = await api.create_filter(user_id=self.mxid, filter_params=fjson)
-        return resp["filter_id"]
+        path = "/_matrix/client/r0/user/" + self.mxid + "/filter"
+        headers = {"Authorization": "Bearer " + api.token}
+        resp = await api.send(method="post", path=path, data=fjson, headers=headers)
+
+        resp_json = await resp.json()
+        return resp_json["filter_id"]
 
     async def connect(self):
         """Create connection object with chat library."""
-        session = aiohttp.ClientSession(trust_env=True)
-        mapi = AsyncHTTPAPI(self.homeserver, session)
+        config = nio.ClientConfig(encryption_enabled=False, pickle_key="")
+        mapi = nio.AsyncClient(self.homeserver, self.mxid, config=config)
 
-        self.session = session
         login_response = await mapi.login(
-            "m.login.password", user=self.mxid, password=self.password
+            password=self.password, device_name="opsdroid"
         )
```

Comment:
I think it would be better to not hard code the device name, maybe we could use the nick from the config instead

Reply:
I think this is a sane default, but we might want to consider allowing it to be configured. In general with bot accounts I think you are only likely to have the single device.

Reply:
Yeah that seems like a better way to do it